homeassistant/components/cloud/google_config.py

```diff
@@ -173,34 +176,48 @@ def _migrate_google_entity_settings_v1(self):
             # Don't migrate if there's a YAML config
             return
 
-        entity_registry = er.async_get(self.hass)
-
-        for entity_id, entry in entity_registry.entities.items():
-            if CLOUD_GOOGLE in entry.options:
-                continue
-            options = {"should_expose": self._should_expose_legacy(entity_id)}
-            if _2fa_disabled := (self._2fa_disabled_legacy(entity_id) is not None):
-                options[PREF_DISABLE_2FA] = _2fa_disabled
-            entity_registry.async_update_entity_options(
-                entity_id, CLOUD_GOOGLE, options
+        for state in self.hass.states.async_all():
```

Comment:
Same Q as for alexa, should we also run through all entities that are known to cloud preferences but are not currently in the state machine.